hot-desk/src/main/java/com/exadel/sandbox/security/service/UserPrincipal.java

```diff
@@ -0,0 +1,54 @@
+package com.exadel.sandbox.security.service;
+
+import com.exadel.sandbox.employee.entity.Employee;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserPrincipal implements UserDetails {
+    private Employee employee;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return null;
+    }
+
+    @Override
+    public String getPassword() {
+        return employee.getPassword();
+    }
+
+    @Override
+    public String getUsername() {
+        return employee.getUsername();
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return false;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return false;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return false;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return false;
```

Comment:
Do u sure that most of all methods should return false and null?

Reply:
I think we don't need these kind of methods which are returning false  or null,  Because we are implementing from UserDetails, we are forced to override them